ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs

```diff
@@ -148,12 +151,20 @@ chainSyncServerForFollower tracer chainDB flr =
 -------------------------------------------------------------------------------}
 
 -- | Events traced by the Chain Sync Server.
---
--- The whole headers/blocks in the traced 'ChainUpdate' are substituted with
--- their corresponding 'Point'.
-data TraceChainSyncServerEvent blk
-  = TraceChainSyncServerRead        (Tip blk) (ChainUpdate blk (Point blk))
-  | TraceChainSyncServerReadBlocked (Tip blk) (ChainUpdate blk (Point blk))
-  | TraceChainSyncRollForward       (Point blk)
-  | TraceChainSyncRollBackward      (Point blk)
+data TraceChainSyncServerEvent blk =
+    -- | Read a chain update instruction non-blockingly.
+    TraceChainSyncServerRead        (TraceChainSyncServerInfo blk)
+    -- | Read a chain update instruction blockingly.
+  | TraceChainSyncServerReadBlocked (TraceChainSyncServerInfo blk)
```

Comment:
Do we want to make `TraceChainSyncServerReadBlocked` just a boolean-isomorph field in `TraceChainSyncServerRead` while we're at it?